Tests/scripts/common.py

```diff
@@ -217,3 +219,106 @@ def replace_escape_characters(sentence: str, replace_with: str = " ") -> str:
     for escape_char in escape_chars:
         sentence = sentence.replace(escape_char, replace_with)
     return sentence
+
+
+def get_pipelines_and_commits(gitlab_url:str, gitlab_access_token:str, project_id:str, look_back_hours:int):
+    """
+    Get all pipelines and commits on the master branch in the last X hours,
+    pipelines are filtered to only include successful and failed pipelines.
+    Args:
+        gitlab_url - the url of the gitlab instance
+        gitlab_access_token - the access token to use to authenticate with gitlab
+        project_id - the id of the project to query
+        look_back_hours - the number of hours to look back for commits and pipeline
+    Return:
+        a list of gitlab pipelines and a list of gitlab commits in ascending order
+    """
+    gl = gitlab.Gitlab(gitlab_url, private_token=gitlab_access_token)
+    project = gl.projects.get(project_id)
+
+    # Calculate the timestamp for look_back_hours ago
+    time_threshold = (
+        datetime.utcnow() - timedelta(hours=look_back_hours)).isoformat()
+
+    commits = project.commits.list(all=True, since=time_threshold, order_by='updated_at', sort='asc')
+    pipelines = project.pipelines.list(all=True, updated_after=time_threshold, ref='master',
+                                       source='push', order_by='updated_at', sort='asc')
+
+    # Filter out pipelines that are not done
+    filtered_pipelines = [
+        pipeline for pipeline in pipelines if pipeline.status in ('success', 'failed')]
+
+    return filtered_pipelines, commits
+
+
+def shame(commit):
```

Comment:
```suggestion
def get_commit_author_data(commit):
```

While shame is correct, this is a public repo and should look accordingly. :)

Reply:
fixed